### ISSUE TITLE
New install not updating

### ISSUE THREAD

**Issue by DrDrewZA** [SickChill/SickChill#7356]
So I finally got around to moving my install over to a new machine.  I was still running on the pip version that was manually installed and I needed to update using the pip command at command line.  I recall someone in the past seeing that once I reinstalled using a ready installer it would update within the program properly.  its not doing it.  how can I get it to update properly?

I notice the branch still says pip (I backed up my config and restored it on the new machine - not sure if thats playing into this?)

Otherwise Sickchill is behaving perfectly - just the updates...

TIA

### Before submitting your issue:

Enable debug logging in SickChill settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit: latest win10 ready installer
OS: Win10
Browser: Edge
What you did: Clicked Update when Sickchill informed me there was an update
What happened:  Http500 error
What you expected:  Sickchill to update
Logs:
```
:39:51 INFO :: POSTPROCESSOR-AUTO :: Beginning auto post processing task: C:\Downloads\Sickrage
2021-07-24 08:39:48 INFO :: POSTPROCESSOR :: Auto post processing task for C:\Downloads\Sickrage was added to the queue
AA TypeError: update() missing 1 required positional argument: 'pid'
AA return function()
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\views\index.py", line 190, in async_call
AA result = self.fn(*self.args, **self.kwargs)
AA File "C:\SickChill\Python3\tools\lib\concurrent\futures\thread.py", line 57, in run
AA value = future.result()
AA File "C:\SickChill\Python3\tools\lib\site-packages\tornado\gen.py", line 762, in run
AA results = yield self.async_call(method)
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\views\index.py", line 164, in get
AA yielded = self.gen.throw(*exc_info) # type: ignore
AA File "C:\SickChill\Python3\tools\lib\site-packages\tornado\gen.py", line 769, in run
AA result = await result
AA File "C:\SickChill\Python3\tools\lib\site-packages\tornado\web.py", line 1704, in _execute
AA Traceback (most recent call last):
AA During handling of the above exception, another exception occurred:
AA TypeError: can only concatenate str (not "Version") to str
AA self._notify_pht(update_text + new_version, title)
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\oldbeard\notifiers\plex.py", line 71, in notify_git_update
AA n.notify_git_update(new_version)
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\oldbeard\notifiers\__init__.py", line 115, in notify_git_update
AA notifiers.notify_git_update(settings.CUR_COMMIT_HASH or "")
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\update_manager\pip.py", line 80, in update
AA return self.updater.update()
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\update_manager\runner.py", line 308, in update
AA if updater.update():
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\views\home.py", line 687, in update
AA return function(**kwargs)
AA File "C:\SickChill\Python3\tools\lib\site-packages\sickchill\views\index.py", line 188, in async_call
AA Traceback (most recent call last):
AA HTTPServerRequest(protocol='http', host='127.0.0.1:8081', method='GET', uri='/home/update/?pid=3852', version='HTTP/1.1', remote_ip='127.0.0.1')
2021-07-24 08:39:38 ERROR :: WEBSERVER :: Uncaught exception GET /home/update/?pid=3852 (127.0.0.1)
2021-07-24 08:29:51 INFO :: POSTPROCESSOR-AUTO :: Auto post processing task for C:\Downloads\Sickrage completed
2021-07-24 08:2
```

**Comment by telsum** [comment]
Hi, I saw this as well. 
I had the same issue with my Windows install.
Re-installing from the current windows installer solved this, however I posted in
https://github.com/SickChill/SickChill/issues/7307#issuecomment-885999344
My git link looks broken in sickchill but that might not be the issue. 
I'd recommend as a short fix backup sickchill via the gui, uninstall and reinstall (if you don't delete the data files, an option on uninstall, you should not need the backup. The backup is a just in case so you can restore it if need be.)

**Comment by Yoshi900** [comment]
I am getting the same issue too. 
All working fine but cannot find the update. Tired uninstall and reinstall and the issue goes away for about 24 hours but then returns. 
my config file is 
git_reset = 1
  branch = pip
  git_remote = origin
  git_remote_url = https://github.com/SickChill/SickChill.git
  cur_commit_hash = 7f07d61b0a8fb34a9d14748829aacd9f9cf90220
  cur_commit_branch = ""
  config_version = 8

**Comment by miigotu** [comment]
> Hi, I saw this as well. 
> I had the same issue with my Windows install.
> Re-installing from the current windows installer solved this, however I posted in
> https://github.com/SickChill/SickChill/issues/7307#issuecomment-885999344
> My git link looks broken in sickchill but that might not be the issue. 
> I'd recommend as a short fix backup sickchill via the gui, uninstall and reinstall (if you don't delete the data files, an option on uninstall, you should not need the backup. The backup is a just in case so you can restore it if need be.)
> 
> I just noticed, you were a PIP install. 
> By per chance does your config.ini say 
> branch = pip  ?? 
> I'm hazarding a guess it will be. 

Windows is pip install now too.

**Comment by miigotu** [comment]
> I am getting the same issue too. 
> All working fine but cannot find the update. Tired uninstall and reinstall and the issue goes away for about 24 hours but then returns. 
> my config file is 
> git_reset = 1
>   branch = pip
>   git_remote = origin
>   git_remote_url = https://github.com/SickChill/SickChill.git
>   cur_commit_hash = 7f07d61b0a8fb34a9d14748829aacd9f9cf90220
>   cur_commit_branch = ""
>   config_version = 8
> 

Git doesn't matter if your branch is pip. If it is pip, you are installed via pip.

**Comment by miigotu** [comment]
There IS a bug with the updater when SC is installed with pip. Be patient.

**Comment by Yoshi900** [comment]
Yeah no problem. The current version is not breaking anything so I will just wait. Just making sure that others have the correct solution too...

**Comment by DrDrewZA** [comment]
> There IS a bug with the updater when SC is installed with pip. Be patient.

Ok cool, can I update using the command line in the meantime without breaking anything?  

Before this move I was using pip3 install -U sickchill - Would that syntax still apply?

**Comment by miigotu** [comment]
Yes

**Comment by Smokey23** [comment]
@miigotu, 

As a user on windows 7, who has only installed via the installer, and was recently moved from GIT to PIP in the background during a reinstall, how do I update from the command line?

I have not installed python or PIP outside of the Sickchill installer.

I searched my computer for PIP3 and found it in the following location:
C:\Users\XXXXX\AppData\Local\Programs\Python\Python36-32\Scripts\

If I open a command prompt and navigate to my sickchill location of C:\SickChill, typing `pip3 install -U sickchill` I get a response of `'pip3' is not recognized as an internal or external command`.

Do I need to install Python by itself or run some other commands in order to update through pip from the command line?  I just don't want to mess my setup up by doing things I shouldn't.

Thanks for all your hard work!

**Comment by cbinder4** [comment]
Try going to your SickChill dir and typing -
C:\Users\XXXXX\AppData\Local\Programs\Python\Python36-32\Scripts\pip3 install –U sickchill

**Comment by DrDrewZA** [comment]
> @miigotu,
> 
> As a user on windows 7, who has only installed via the installer, and was recently moved from GIT to PIP in the background during a reinstall, how do I update from the command line?
> 
> I have not installed python or PIP outside of the Sickchill installer.
> 
> I searched my computer for PIP3 and found it in the following location:
> C:\Users\XXXXX\AppData\Local\Programs\Python\Python36-32\Scripts\
> 
> If I open a command prompt and navigate to my sickchill location of C:\SickChill, typing `pip3 install -U sickchill` I get a response of `'pip3' is not recognized as an internal or external command`.
> 
> Do I need to install Python by itself or run some other commands in order to update through pip from the command line? I just don't want to mess my setup up by doing things I shouldn't.
> 
> Thanks for all your hard work!

You have a couple of options:

1)  you could add the dir pipe.exe is in to your path in the environment variables.  Then it will run from anywhere and let you update.
2)  if this isn't something you know how or want to do you could install the latest version of python and select the add to path checkbox during installation.  Then it will add it to the path for you and pip will run from anywhere.
3)  use the suggestion cbinder made, but 8bthink you'll need to add .exe to the end of pip in order order that to work.

**Comment by miigotu** [comment]
> @miigotu, 
> 
> As a user on windows 7, who has only installed via the installer, and was recently moved from GIT to PIP in the background during a reinstall, how do I update from the command line?
> 
> I have not installed python or PIP outside of the Sickchill installer.
> 
> I searched my computer for PIP3 and found it in the following location:
> C:\Users\XXXXX\AppData\Local\Programs\Python\Python36-32\Scripts\
> 
> If I open a command prompt and navigate to my sickchill location of C:\SickChill, typing `pip3 install -U sickchill` I get a response of `'pip3' is not recognized as an internal or external command`.
> 
> Do I need to install Python by itself or run some other commands in order to update through pip from the command line?  I just don't want to mess my setup up by doing things I shouldn't.
> 
> Thanks for all your hard work!

iirc:

C:\SickaChill\Python3\tools\python -m pip -U sickchill

**Comment by mwhitmel** [comment]
> There IS a bug with the updater when SC is installed with pip. Be patient.

Hi,
Im in the same position the PIP install no longer updating. I see this is being worked on. Will there be a noticed so we know when it is fixed ?

thanks

**Comment by davidreverett** [comment]
If you don't want to wait for the update to be fixed, you can do the following- I tried it and it worked. 

1. Uninstall SC- it is important that you either backup your config first, and/or be sure when the installer asks if you want to remove everything to say NO- you are planning to reinstall. 
2. Reinstall using the current Windows Installer- it'll download and install the updated version automatically. 
3. * (If you are using NAS storage or other shared media, set the Sickchill service in Windows Services Control Panel to run as the appropriate account that has rights to the shared media, and restart the service after making the change.)

Hope that helps- it got me from 7.14 to 7.23 and I now no longer have the broken Update Alert at the top of my screen.

**Comment by davidreverett** [comment]
Not stale

**Comment by miigotu** [comment]
This was fixed in 75f0e4820c232422124ab8b158313d78231fbae1
